source/simulator/stokes_matrix_free.cc

```diff
@@ -1703,6 +1779,111 @@ namespace aspect
             level_cell_data[level].enable_newton_derivatives = false;
         }
     }
+
+    {
+      // Create active mesh tables to store the product of the pressure perturbation and
+      // the normalized gravity used in the free surface stabilization.
+      // Currently, mutilevel is not implemented yet, it may slow down the convergence.
+
+      // TODO: implement multilevel surface terms for the free surface stabilization.
+
+      active_cell_data.apply_stabilization_free_surface_faces = false;
+
+      if (sim.mesh_deformation
+          && !sim.mesh_deformation->get_free_surface_boundary_indicators().empty())
+        {
+          const double free_surface_theta =
+            sim.mesh_deformation->template get_matching_mesh_deformation_object<MeshDeformation::FreeSurface<dim>>()
+          .get_free_surface_theta();
+
+          // GMG doesn't support melt transport
+          AssertThrow(!sim.parameters.include_melt_transport, ExcNotImplemented());
```

Comment:
Flip these two lines. Checking for impossible conditions should go before anything you do.

Also shouldn't this go into the constructor near line 1368? Or is this parameter not yet set at that time?